### PR TITLE
docs: Add notes on artifact streaming

### PR DESCRIPTION
**Pull request by terrytangyuan**
Added some docs related to https://github.com/argoproj/argo-workflows/issues/8489 https://github.com/argoproj/argo-workflows/pull/11823

### CODE REVIEW

File: docs/configure-artifact-repository.md [hunk 1]
```diff
@@ -554,3 +554,12 @@ configuring the default artifact repository described previously.
       command: [sh, -c]
       args: ["cp -r /my-input-artifact /my-output-artifact"]
 ```
+
+## Artifact Streaming
+
+With artifact streaming, artifacts don’t need to be saved to disk first. Artifact streaming is only supported in the following
+artifact drivers: S3 (v3.4+), HTTP (v3.5+), and Artifactory (v3.5+).
```
Comment (juliev0) [inline docs/configure-artifact-repository.md:561]: I see it here too: https://github.com/argoproj/argo-workflows/blob/master/workflow/artifacts/azure/azure.go#L208

File: docs/configure-artifact-repository.md [hunk 2]
```diff
@@ -554,3 +554,12 @@ configuring the default artifact repository described previously.
...
+
+Previously, when a user clicks the download artifact button in the UI, the artifact needs to be written to the
+Argo server’s disk first before downloading. If many users are trying to download at the same time, they will take up
```
Comment (juliev0) [inline docs/configure-artifact-repository.md:564]: nitpick: "Previously, when a user _would click_ the _button to download an artifact_ in the UI, the artifact _would need_ to be written"...

Comment (terrytangyuan) [reply docs/configure-artifact-repository.md]: Thanks!

File: docs/configure-artifact-repository.md (hunk 1 above)
Comment (terrytangyuan) [reply docs/configure-artifact-repository.md]: Added!

File: docs/configure-artifact-repository.md [hunk 3]
```diff
@@ -558,8 +558,8 @@ configuring the default artifact repository described previously.
 ## Artifact Streaming
 
 With artifact streaming, artifacts don’t need to be saved to disk first. Artifact streaming is only supported in the following
-artifact drivers: S3 (v3.4+), HTTP (v3.5+), and Artifactory (v3.5+).
+artifact drivers: S3 (v3.4+), Azure Blob (v3.4+), HTTP (v3.5+), and Artifactory (v3.5+).
 
-Previously, when a user clicks the download artifact button in the UI, the artifact needs to be written to the
-Argo server’s disk first before downloading. If many users are trying to download at the same time, they will take up
+Previously, when a user would click the button to download an artifact in the UI, the artifact would need to be written to the
+Argo Server’s disk first before downloading. If many users try to download simultaneously, they will take up
```
Comment (juliev0) [inline docs/configure-artifact-repository.md:564]: sorry I didn't notice this the first time! but can we change the verb tense here too? "If many users tried to download simultaneously, they would take up"

Comment (terrytangyuan) [reply docs/configure-artifact-repository.md]: No problem. Updated.

**Review by juliev0 — APPROVED** [review]

**Pull request merged**